consensus/istanbul/backend/backend_test.go

```diff
@@ -0,0 +1,352 @@
+// Copyright 2019 The klaytn Authors
+// This file is part of the klaytn library.
+//
+// The klaytn library is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Lesser General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// The klaytn library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public License
+// along with the klaytn library. If not, see <http://www.gnu.org/licenses/>.
+
+package backend
+
+import (
+	"github.com/hashicorp/golang-lru"
+	"github.com/klaytn/klaytn/blockchain/types"
+	"github.com/klaytn/klaytn/common"
+	"github.com/klaytn/klaytn/consensus/istanbul"
+	istanbulCore "github.com/klaytn/klaytn/consensus/istanbul/core"
+	"github.com/klaytn/klaytn/consensus/istanbul/validator"
+	"github.com/klaytn/klaytn/crypto"
+	"github.com/klaytn/klaytn/event"
+	"github.com/klaytn/klaytn/governance"
+	"github.com/klaytn/klaytn/networks/p2p"
+	"github.com/klaytn/klaytn/params"
+	"github.com/klaytn/klaytn/reward"
+	"github.com/klaytn/klaytn/storage/database"
+	"math/big"
+	"testing"
+)
+
+var (
+	// testing node's private key
+	PRIVKEY = "ce7671a2880493dfb8d04218707a16b1532dfcac97f0289d770a919d5ff7b068"
+	// testing node should be in a committee in these blocks
+	committeeBlocks = []int64{0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 20, 21, 22}
+	// Max blockNum
+	maxBlockNum = int64(100)
+)
+
+func getTestCouncil() []common.Address {
+	return []common.Address{
+		common.HexToAddress("0x414790CA82C14A8B975cEBd66098c3dA590bf969"), // Node Address for test
```

Comment:
Are these pre-generated addresses needed for the test?

Reply:
@ehnuje Actually, it can be generated programmatically. But I worried that some kind of hard fork may happen even though it passed the test. I think testing with a hard coded data based test result would be more helpful to know if it make some hard fork status.

Reply:
I'm sorry but I could hardly think about the situation when this data affects the test when a hard fork happens. Could you please explain in a bit detail?

Reply:
@ehnuje My idea is simple. If the data which passed before can't pass the test now, it is clear that there is some problem. If the test data is generated dynamically and the test would fail, it will be much tough to know what is a problem. And it also make harder to determine the result of `SubList()` is right or not.

Reply:
Thanks for the explanation. I understood your point.